src/main/java/com/ait/lienzo/client/core/shape/TextBoundsWrap.java

```diff
@@ -0,0 +1,136 @@
+/*
+   Copyright (c) 2017 Ahome' Innovation Technologies. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape;
+
+import java.util.ArrayList;
+
+import com.ait.lienzo.client.core.Context2D;
+import com.ait.lienzo.client.core.types.BoundingBox;
+import com.ait.lienzo.client.core.types.TextMetrics;
+import com.ait.lienzo.client.core.util.ScratchPad;
+import com.ait.lienzo.shared.core.types.ColorName;
+import com.ait.lienzo.shared.core.types.TextAlign;
+import com.ait.lienzo.shared.core.types.TextBaseLine;
+import com.ait.lienzo.shared.core.types.TextUnit;
+import com.ait.tooling.nativetools.client.collection.NFastDoubleArrayJSO;
+import com.google.gwt.canvas.dom.client.CanvasPixelArray;
+
+/**
+ * Text implementation for Canvas.
+ */
+public class TextBoundsWrap extends TextNoWrap
+{
+
+    private BoundingBox wrapBoundaries;
+
+    public TextBoundsWrap(final ITextWrapper.Supplier<String> textSupplier,
+                          final ITextWrapper.Supplier<Double> fontSizeSupplier,
+                          final ITextWrapper.Supplier<String> fontStyleSupplier,
+                          final ITextWrapper.Supplier<String> fontFamilySupplier,
+                          final ITextWrapper.Supplier<TextUnit> textUnitSupplier,
+                          final ITextWrapper.Supplier<TextBaseLine> textBaseLineSupplier,
+                          final ITextWrapper.Supplier<TextAlign> textAlignSupplier,
+                          final BoundingBox wrapBoundaries) {
+        super(textSupplier,
+              fontSizeSupplier,
+              fontStyleSupplier,
+              fontFamilySupplier,
+              textUnitSupplier,
+              textBaseLineSupplier,
+              textAlignSupplier);
+        this.wrapBoundaries=wrapBoundaries;
+    }
+
+    public BoundingBox getWrapBoundaries() {
+        return wrapBoundaries;
+    }
+
+    public TextBoundsWrap setWrapBoundaries(final BoundingBox boundaries) {
+        wrapBoundaries = boundaries;
+        return this;
+    }
+
+    @Override
+    public BoundingBox getBoundingBox()
+    {
+        double width = wrapBoundaries.getWidth();
+        int numOfLines = 1;
+        String[] words = textSupplier.get().split(" ");
+        StringBuilder nextLine = new StringBuilder(words[0]);
```

Comment:
There is an issue with my previous implementation: if you supplied it a string consisting of only spaces, then split returns a 0 elements array (so `words[0]` does not exist). Add an array length check here.